### PR TITLE
Mass purrbation affects mutant species

### PR DESCRIPTION
**Pull request by ghost**
aaa
Mass purrbation gives cat organs to other species.

## Why It's Good For The Game
Increases salt harvest from adminbus.
## Changelog
:cl:
admin: Mass-Purrbation now affects mutant species. Nya!
/:cl:

### CODE REVIEW

**Comment by shellspeed1** [comment]
cursed

**Comment by Tharcoonvagh** [comment]
intended

**Comment by Time-Green** [comment]
NNNyyyaaa~~~

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 1]
```diff
@@ -131,17 +131,31 @@
...
-	if(!ishuman(H) || !iscatperson(H))
-		return
-
-	H.set_species(/datum/species/human)
-
+	if(ishumanbasic(H) || iscatperson(H))
+		H.set_species(/datum/species/human)
+		H.visible_message("[H] if 1")
+	else if(ishuman(H))
+		H.visible_message("[H] if 2")
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: Nice debug message.

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 2]
```diff
@@ -131,17 +131,31 @@
...
 		playsound(get_turf(H), 'sound/effects/meow1.ogg', 50, TRUE, -1)
 
 /proc/purrbation_remove(mob/living/carbon/human/H, silent = FALSE)
-	if(!ishuman(H) || !iscatperson(H))
-		return
-
-	H.set_species(/datum/species/human)
-
+	if(ishumanbasic(H) || iscatperson(H))
+		H.set_species(/datum/species/human)
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: Why turn basic humans into basic hoomans??

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 3]
```diff
@@ -131,17 +131,31 @@
...
+		H.visible_message("[H] if 1")
+	else if(ishuman(H))
+		H.visible_message("[H] if 2")
+		var/organs = H.internal_organs
+		for(var/obj/item/organ/current_organ in organs)
+			if(istype(current_organ, /obj/item/organ/tail/cat))
+				H.visible_message("[H] if 3")
+				current_organ.Remove(H, TRUE)
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: If you wanna be extra fancy, you can give lizzies their tails back.

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 4]
```diff
@@ -131,17 +131,31 @@
...
+			if(istype(current_organ, /obj/item/organ/ears/cat))
+				H.visible_message("[H] if 4")
+				var/obj/item/organ/ears/new_ears = new
```
Comment (nemvar) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: Same with custom mutant ears.

File: code/modules/mob/living/carbon/human/species_types/felinid.dm (hunk 3 above)
Comment (py01) [reply code/modules/mob/living/carbon/human/species_types/felinid.dm]: will work on

**Comment by spessbandit** [comment]
critical fix speedmerge immediately

**Comment by PKPenguin321** [comment]
dude, gross

**Comment by mrhugo13** [comment]
### No.
**No.**
_No._
**_No._**
> No.
`No.`
[No.](https://cdn.discordapp.com/attachments/501332410835795968/627904136964472833/democringe.mp4)
- No
1. No
- [ ] No

**Comment by Shadowflame909** [comment]
No

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 5]
```diff
@@ -131,17 +131,31 @@
...
 		playsound(get_turf(H), 'sound/effects/meow1.ogg', 50, TRUE, -1)
 
 /proc/purrbation_remove(mob/living/carbon/human/H, silent = FALSE)
-	if(!ishuman(H) || !isfelinid(H))
-		return
-
-	H.set_species(/datum/species/human)
-
+	if(isfelinid(H))
+		H.set_species(/datum/species/human)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: is it intentional that purrbation removes people who were originally felinid? seems like there needs to be a check for that somewhere.

Comment (py01) [reply code/modules/mob/living/carbon/human/species_types/felinid.dm]: It was like that before, but I will add that

**Comment by PKPenguin321** [comment]
No joke whenever a shitmin pushes this awful button I suicide and change servers

**Pull request merged**